src/Sylius/Bundle/CoreBundle/Controller/OrderController.php

```diff
@@ -111,4 +124,159 @@ public function historyAction(Request $request)
 
         return $this->viewHandler->handle($configuration, $view);
     }
+
+    /**
+     * @param Request $request
+     * @param int $orderId
+     *
+     * @return Response
+     */
+    public function payAction(Request $request, $orderId)
```

Comment:
I think that it would be a nice and clean none resource controller, but i agree with you.